File: vbnet/prime_number_checker/prime_number_checker_9.vb
' This program is a prime number checker written in Visual Basic .NET.
' The purpose of this program is to determine whether a given number is a prime number.
' A prime number is a natural number greater than 1 that has no positive divisors other than 1 and itself.

Module PrimeNumberChecker

    ' Main subroutine that serves as the entry point of the program
    Sub Main()
        ' Declare a variable to store the number to be checked
        Dim number As Integer

        ' Prompt the user to enter a number
        Console.WriteLine("Enter a number to check if it is a prime number:")
        number = Convert.ToInt32(Console.ReadLine())

        ' Call the function to check if the number is prime and store the result
        Dim isPrime As Boolean = IsPrimeNumber(number)

        ' Display the result to the user
        If isPrime Then
            Console.WriteLine("The number " & number & " is a prime number.")
        Else
            Console.WriteLine("The number " & number & " is not a prime number.")
        End If

        ' Wait for user input before closing the console window
        Console.ReadLine()
    End Sub

    ' Function to check if a number is prime
    Function IsPrimeNumber(ByVal num As Integer) As Boolean
        ' A prime number must be greater than 1
        If num <= 1 Then
            Return False
        End If

        ' Check for factors other than 1 and the number itself
        For i As Integer = 2 To num - 1
            ' If a factor is found, the number is not prime
            If num Mod i = 0 Then
                Return False
            End If
        Next

        ' If no factors are found, the number is prime
        Return True
    End Function

End Module

File: rust/prime_number_checker/prime_number_checker_9.rs
// This program is designed to check if a given number is a prime number.
// It includes detailed comments to help programmers understand the flow of the program.
// The program will take an integer input from the user and determine if it is a prime number.
// Prime numbers are numbers greater than 1 that have no divisors other than 1 and themselves.

use std::io;

// Function to check if a number is prime
fn is_prime(n: i32) -> bool {
    // A prime number is greater than 1
    if n <= 1 {
        return false;
    }

    // Check for factors from 2 to n-1
    for i in 2..n {
        if n % i == 0 {
            return false;
        }
    }

    true
}

fn main() {
    // Variable to store user input
    let mut input = String::new();

    // Prompt the user for input
    println!("Enter a number to check if it is prime:");

    // Read the input from the user
    io::stdin().read_line(&mut input).expect("Failed to read line");

    // Convert the input to an integer
    let number: i32 = input.trim().parse().expect("Please type a number!");

    // Check if the number is prime
    let result = is_prime(number);

    // Print the result
    if result {
        println!("The number {} is a prime number.", number);
    } else {
        println!("The number {} is not a prime number.", number);
    }

    // Additional unused variables and functions for verbosity
    let weather = "sunny";
    let temperature = 25;
    let mut unused_variable = 42;

    fn unused_function() {
        let x = 10;
        let y = 20;
        let z = x + y;
        println!("This is an unused function: {}", z);
    }

    unused_variable = 100;
    println!("The weather is {} and the temperature is {} degrees.", weather, temperature);
}

